src/model/window-guardian.ts

```diff
@@ -68,25 +68,21 @@ const makeWindowGuardianConfig = (
 
 export const makeGuardianBrowserCAPI = (CAPI: CAPIType): CAPIBrowserType => {
 	// For some elements it is important to keep thier index in the `elements` array
-
 	const blockElementWithIndex = <T extends CAPIElement>(
 		blocks: { elements: CAPIElement[] }[],
 		blockElementType: CAPIElement['_type'],
 		indexName: string,
 	): T[] => {
 		if (blocks.length === 0) return [];
-		return blocks[0].elements.reduce(
-			(acc: T[], element: CAPIElement, index: number) => {
-				if (element._type === blockElementType) {
-					acc.push({
-						...element,
-						[indexName]: index,
-					} as T);
-				}
-				return acc;
-			},
-			[] as T[],
-		);
+		return blocks[0].elements.map((element: CAPIElement, index: number) => {
+			if (element._type === blockElementType) {
+				return {
+					...element,
+					[indexName]: index,
+				};
+			}
+			return element;
```

Comment:
I'm not sure this will have the same behaviour as before will it? This is leaving elements unchanged if they aren't of `blockElementType` whereas the reduce actually filtered them out.  I think we'd need to do a flatMap / filter then map. I do think that would be easier to read than the reduce though.

Reply:
That's a great catch. `filter` and then `map` would definitely be more readable because the logic follows the function purposes

Reply:
I've made it `map` then `filter`, as per our discussion on chat. Thanks for helping here!

Reply:
Was this due to type predicate, as we’re running into #4582?